### PR TITLE
chore(txpool): implement ensure_max_init_code_size()

### LINKED ISSUE #924: Add `maxInitCodeSize` txpool test
### Describe the feature

An edge case in geth's txpool implementation was recently found, and fixed in [go-ethereum#26504](https://github.com/ethereum/go-ethereum/pull/26504). We should create a test for this case.

The intended behavior of the txpool is to reject transactions where the transaction is a contract creation and the `tx.data` exceeds `maxInitCodeSize`.

The `maxInitCodeSize` and `maxCodeSize`, a related constant, [are listed in geth here](https://github.com/ethereum/go-ethereum/blob/a35b654f25fdac518d5775b55aa064025040b01a/params/protocol_params.go#L127-L128):
```go
	MaxCodeSize     = 24576           // Maximum bytecode to permit for a contract
	MaxInitCodeSize = 2 * MaxCodeSize // Maximum initcode to permit in a creation transaction and create instructions
```

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by jinsankim**
Closes #924 

I'm not much sure whether it could close #924. I'll be really appreciate to point out me if I misunderstood the issue.

### CODE REVIEW

File: crates/transaction-pool/src/traits.rs [hunk 1]
```diff
@@ -348,6 +352,12 @@ impl PoolTransaction for PooledTransaction {
         }
     }
 
+    /// Returns the transaction's [`TransactionKind`], which is the address of the recipient or
+    /// [`TransactionKind::Create`] if the transaction is a contract creation.
+    fn kinds(&self) -> &TransactionKind {
```
Comment (mattsse) [inline crates/transaction-pool/src/traits.rs:357]:
```suggestion
    fn kind(&self) -> &TransactionKind {
```

File: crates/transaction-pool/src/validate.rs [hunk 2]
```diff
@@ -54,6 +54,31 @@ pub trait TransactionValidator: Send + Sync {
...
         transaction: Self::Transaction,
     ) -> TransactionValidationOutcome<Self::Transaction>;
+
+    /// Ensure that the code size is not greater than `max_init_code_size`.
+    /// `max_init_code_size` should be configurable so this will take it as an argument.
+    fn ensure_max_init_code_size(
+        &self,
+        transaction: Self::Transaction,
+        max_init_code_size: usize,
+    ) -> Result<(), PoolError> {
```
Comment (mattsse) [inline crates/transaction-pool/src/validate.rs:64]: this seems ok for now, we'll figure out the right ergonomics later

**Review by mattsse — CHANGES_REQUESTED** [review]
smol nit, otherwise lgtm

File: crates/transaction-pool/src/traits.rs (hunk 1 above)
Comment (jinsankim) [reply crates/transaction-pool/src/traits.rs]: Thanks for comment. I follow up the change request w/ edfaa6c12bfb78882a88c3cde97b14dcad7c70a5.

**Review by mattsse — APPROVED** [review]
